plugins/modules/mso_backup.py

```diff
@@ -278,10 +278,19 @@ def main():
             mso.existing = mso.proposed
         else:
             try:
-                payload = dict(name=(os.path.basename(backup), open(backup, "rb"), "application/x-gzip"))
-                mso.existing = mso.request_upload("backups/upload", fields=payload)
-            except Exception:
-                mso.module.fail_json(msg="Backup file '{0}' not found!".format(", ".join(backup.split("/")[-1:])))
+                request_url = "backups/upload"
+                payload = dict()
+                if mso.platform == "nd":
+                    if remote_location is None or remote_path is None:
+                        mso.module.fail_json(msg="NDO backup upload failed: remote_location and remote_path is required for NDO backup upload")
```

Comment:
```suggestion
                        mso.module.fail_json(msg="NDO backup upload failed: remote_location and remote_path are required for NDO backup upload")
```

Reply:
Can you even reach this condition with the required_if above?

Reply:
The remote_path  is optional for the older MSO platform, so removed remote_path from required_if.